src/components/TaskVariablesModal.vue

```diff
@@ -0,0 +1,172 @@
+<template>
+  <transition name="modal">
+    <div class="modal-mask">
+      <div class="modal-wrapper">
+        <div class="modal-container" :style="{ 'background-color': backgroundColor }">
+
+          <div class="modal-header">
+              <h1 class="header-title">Settings</h1>
+          </div>
+
+          <div class="modal-body">
+            <slot v-for="taskVariableKey in Object.keys(taskVariables)" name="body">
+              <label :key="taskVariableKey + 'label'" :for="taskVariableKey + 'input'">{{ toFreeText(taskVariableKey) }}:</label>
+              <input v-if="taskVariables[taskVariableKey].type === 'string' || taskVariables[taskVariableKey].type === 'lambda'" :key="taskVariableKey + 'input'" v-model="taskVariables[taskVariableKey].value" type="text">
+              <input v-if="taskVariables[taskVariableKey].type === 'float' || taskVariables[taskVariableKey].type === 'duration'" :key="taskVariableKey + 'input'" v-model.number="taskVariables[taskVariableKey].value" type="number">
+            </slot>
+          </div>
+
+          <div class="modal-footer">
+            <slot name="footer">
+              <button class="modal-default-button" @click="saveTaskVariables">
+                Save
+              </button>
+            </slot>
+          </div>
+        </div>
+      </div>
+    </div>
+  </transition>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { Component, Prop } from 'vue-property-decorator';
+
+import Alert from '@/interfaces/Alert';
+import KapacitorApi from '@/apis/KapacitorApi.ts';
+
+@Component
+export default class TaskVariablesModal extends Vue {
+  // region properties
+  @Prop(String) public taskName!: string;
+  @Prop(String) public backgroundColor!: string;
+  // endregion
+
+  // region public members
+  // endregion
+
+  // region public methods
+  // endregion
+
+  // region private members
+  private taskVariables: any = {};
+  // endregion
+
+  // region constructor
+  // endregion
+
+  // region private methods
+  private mounted() {
+    this.fetchTaskVariables();
+  }
+
+  private toFreeText(text: string): string {
+    const result = text.replace(/([A-Z])/g, ' $1');
+    const finalResult = result.charAt(0).toUpperCase() + result.slice(1);
+
+    return finalResult;
+  }
+
+  private async fetchTaskVariables(): Promise<void> {
+    this.taskVariables = await KapacitorApi.taskVariables(this.taskName);
+  }
+
+  private async saveTaskVariables(): Promise<void> {
+    this.$emit('close');
+
+    await KapacitorApi.updateTaskVariables(this.taskName, this.taskVariables);
+  }
+  // endregion
+}
+</script>
+
+<style lang="less">
+
```

Comment:
Is this your code or do we need to state a reference to someone?

Reply:
https://vuejs.org/v2/examples/modal.html